pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/TNCudaBase.hpp

```diff
@@ -60,133 +57,194 @@ class TNCudaBase : public TensornetBase<PrecisionT, Derived> {
     using CFP_t = decltype(cuUtil::getCudaType(PrecisionT{}));
     using ComplexT = std::complex<PrecisionT>;
     using BaseType = TensornetBase<PrecisionT, Derived>;
-    SharedTNCudaHandle handle_;
+
+  protected:
+    // Note both maxBondDim_ and bondDims_ are used for both MPS and Exact
+    // Tensor Network. Per Exact Tensor Network, maxBondDim_ is 1 and bondDims_
+    // is {1}. Per Exact Tensor Network, setting bondDims_ allows call to
+    // appendInitialMPSState_() to append the initial state to the Exact Tensor
+    // Network state.
+    const std::size_t
+        maxBondDim_; // maxBondDim_ default is 1 for Exact Tensor Network
+    const std::vector<std::size_t>
+        bondDims_; // bondDims_ default is {1} for Exact Tensor Network
+
+  private:
+    const std::vector<std::vector<std::size_t>> sitesModes_;
+    const std::vector<std::vector<std::size_t>> sitesExtents_;
+    const std::vector<std::vector<int64_t>> sitesExtents_int64_;
```

Comment:
why do we need an extra vector vector? I'm guessing this is for cuda? can we just have one and cast between the two during function calls?

Reply:
yea, this is to meet the data type requirement of `cutensornet`. It could be converted on the fly but it's sub-optimal to me since this vector is not only used to initialize the graph but also append the final state of the graph for mps. Moreover, for shot-meausrement, we have to append the final state multiple times.